File: csharp/lightsabers/polymorphism.cs
"council/if_else.cs"

interface IWieldable
{
  public string EnergyColor();
}

class Jedi : IWieldable
{
  string[] colors = { "green", "blue", "purple" };
  public string EnergyColor()
  {
    return colors[random.Next(0, colors.length)];
  }
}

class Sith : IWieldable
{
  public string EnergyColor()
  {
    return "red";
  }
}

class Lightsaber
{
  public void AfterAdd(IWieldable klass)
  {
    this.color = klass.EnergyColor();
  }
}

File: ruby/lightsabers/polymorphism.rb
class Jedi < ActiveRecord::Base
  has_many :lightsabers, as: :warrior

  def energy_color
    ['green', 'blue', 'purple'].sample
  end
end

class Sith < ActiveRecord::Base
  has_many :lightsabers, as: :warrior

  def energy_color
    'red'
  end
end

class Lightsaber < ActiveRecord::Base
  belongs_to :warrior, polymorphic: :true, after_add: :set_color

  def set_color(warrior)
    self.color = warrior.energy_color
  end
end
